packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,384 @@
+   * @returns an EIP-1186 formatted proof
+   */
+  async getProof(address: Address, storageSlots: Buffer[] = []): Promise<Proof> {
+    const proof = await this.provider.send('eth_getProof', [
+      address.toString(),
+      [storageSlots.map((slot) => bufferToHex(slot))],
+      this.blockTag,
+    ])
+
+    return proof
+  }
+
+  /**
+   * Helper method to retrieve a block from the provider to use in the VM
+   * @param blockTag block hash or block number to be run
+   * @param common Common instance used in VM
+   * @returns the block specified by `blockTag`
+   */
+  getBlockFromProvider = async (blockTag: string | bigint, common: Common) => {
+    let blockData
+    if (typeof blockTag === 'string' && blockTag.length === 66) {
+      blockData = await this.provider.send('eth_getBlockByHash', [blockTag, true])
+    } else if (typeof blockTag === 'bigint') {
+      blockData = await this.provider.send('eth_getBlockByNumber', [bigIntToHex(blockTag), true])
+    } else if (
+      isHexPrefixed(blockTag) ||
+      blockTag === 'latest' ||
+      blockTag === 'earliest' ||
+      blockTag === 'pending'
+    ) {
+      blockData = await this.provider.send('eth_getBlockByNumber', [blockTag, true])
+    } else {
+      throw new Error(
+        `expected blockTag to be block hash, bigint, hex prefixed string, or earliest/latest/pending; got ${blockTag}`
+      )
+    }
+
+    const uncleHeaders = []
+    if (blockData.uncles.length > 0) {
+      for (let x = 0; x < blockData.uncles.length; x++) {
+        const headerData = await this.provider.send('eth_getUncleByBlockHashAndIndex', [
+          blockData.hash,
+          intToHex(x),
+        ])
+        uncleHeaders.push(headerData)
+      }
+    }
+
+    return blockFromRpc(blockData, uncleHeaders, {
+      common,
+    })
+  }
+
+  /**
+   * Checkpoints the current state of the StateManager instance.
+   * State changes that follow can then be committed by calling
+   * `commit` or `reverted` by calling rollback.
+   *
+   * Partial implementation, called from the subclass.
+   */
+  async checkpoint(): Promise<void> {
+    this._cache.checkpoint()
+  }
+
+  /**
+   * Commits the current change-set to the instance since the
+   * last call to checkpoint.
+   *
+   * Partial implementation, called from the subclass.
+   */
+  async commit(): Promise<void> {
+    // setup cache checkpointing
+    this._cache.commit()
+  }
+
+  /**
+   * Reverts the current change-set to the instance since the
+   * last call to checkpoint.
+   *
+   * Partial implementation , called from the subclass.
+   */
+  async revert(): Promise<void> {
+    // setup cache checkpointing
+    this._cache.revert()
+  }
+
+  async flush(): Promise<void> {
+    await this._cache.flush()
+  }
+
+  getStateRoot = async () => {
+    return this.stateRoot ?? setLengthLeft(Buffer.from([]), 32)
+  }
+
+  setStateRoot = async (root: Buffer) => {
+    if (root.length !== 32) throw new Error('Root must contain 32 bytes')
```

Comment:
there should ideally be the check here that this matches with the blockTag since this seems to be on the public api, or could just throw that the stateroot can't be set outside `init/setTag` function